x-pack/legacy/plugins/siem/public/components/page_router/pageroute.tsx

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useEffect } from 'react';
+
+export const PageRoute = (props: { title: string; component: React.ReactType }) => {
+  const { title, ...rest } = props;
+  useEffect(() => {
+    document.title = `${title} - Kibana`;
+  });
```

Comment:
Nice effect!

You're going to want to avoid multiple rendering whenever you do things such as resize the window. With this current implementation it will call `useEffect` and set the title ~7 times on an initial page load and it will do it multiple times when you resize your window if you put a console.log statement to test it:

<img width="343" alt="Screen Shot 2019-07-29 at 4 05 33 PM" src="https://user-images.githubusercontent.com/1151048/62086232-edcc6500-b21a-11e9-8fdb-8cc6cc88845d.png">


The fix is simple, which is to pass an empty array to the `useEffect` as a second argument. That will cause the hook to operate like a `componenetDidMount`

```
  useEffect(() => {
    document.title = `${title} - Kibana`;
  }, []);
```

Reply:
Good catch @FrankHassanabad 👍 , I passed title as the second param, so it should do the effect only when title changed. I think it does the same thing as passing `[]` as the second paraam, it just I think put title here would be clearer to express the idea that this effect acts when title changed.
```
 useEffect(() => {
    document.title = `${title} - Kibana`;
  }, [title]);
```

Reply:
That works as well and adds a better degree of safety! I like it much more. 👍